packages/predicate/src/predicate.ts

```diff
@@ -73,8 +79,15 @@ export class Predicate<ARGS extends InputValue[]> extends Account {
   }
 
   setData<T extends ARGS>(...args: T) {
+    const paddedCode = new ByteArrayCoder(this.bytes.length).encode(this.bytes);
+    const OFFSET =
+      VM_TX_MEMORY +
+      TRANSACTION_SCRIPT_FIXED_SIZE +
+      TRANSACTION_PREDICATE_COIN_FIXED_SIZE +
+      paddedCode.byteLength -
+      17;
```

Comment:
I've set up a Sway project to test the PR implementation and noticed a slight difference in the encoding of the main function's arguments for Predicates compared to Scripts.

I'm sharing the Sway project and the TypeScript test I've used in this [Gist](https://gist.github.com/Torres-ssf/338f93901f041fc2a6cb0eba2b598ee2). 

This is working fine and returning the expected output when is a Script. But turning into a Predicate, it will throw the error related to the "The transaction contains a predicate which failed to validate", even though the expected outcome is the predicate returning true, just like happened when it was a Script. 

Looking at the Predicate data argument binary it's possible to notice a slightly difference when compared to the Script data argument binary, even though they are being encoded using the exact same TS array.

```ts
// predicate
0x0000000000002baf000000000000000200000000000000020000000000002c3f000000000000000300000000000000030000000000002c57000000000000000200000000000000020000000000002c87000000000000000200000000000000020000000000002c97000000000000000300000000000000030000000000002caf000000000000000200000000000000020000000000002cdf0000000000000002000000000000000200000000000000180000000000000038000000000000002b000000000000001d000000000000000c00000000000007e4000000000000000c000000000000000800000000000007e30000000000000000000000000000000100000000000000180000000000000038000000000000002b000000000000001d000000000000000c00000000000007e4000000000000000c000000000000000800000000000007e300000000000000000000000000000001

// script
0x0000000000002af0000000000000000200000000000000020000000000002b80000000000000000300000000000000030000000000002b98000000000000000200000000000000020000000000002bc8000000000000000200000000000000020000000000002bd8000000000000000300000000000000030000000000002bf0000000000000000200000000000000020000000000002c200000000000000002000000000000000200000000000000180000000000000038000000000000002b000000000000001d000000000000000c00000000000007e4000000000000000c000000000000000800000000000007e30000000000000000000000000000000100000000000000180000000000000038000000000000002b000000000000001d000000000000000c00000000000007e4000000000000000c000000000000000800000000000007e300000000000000000000000000000001
```